### PR TITLE
Starts readonly proc for Quay

### PR DESCRIPTION
**Pull request by stevsmit**

### CODE REVIEW

File: modules/optional-enabling-read-only-mode-backup-restore.adoc [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+----
+
+. Next, add the key approval with the following query:
++
+[source,terminal]
+----
+quay=# INSERT INTO servicekeyapproval ('approval_type', 'notes')
+  VALUES ("Super User API",
+           {include_notes_here_on_why_this_is_being_added});
+----
```
Comment (LiZhang19817) [inline modules/optional-enabling-read-only-mode-backup-restore.adoc:143]: In Quay 3.11 the SQL Command is not correct, the name of approval_type should be "ServiceKeyApprovalType.SUPERUSER", here also give example of the command.

```
quayregistry-quay-database=# select * from servicekeyapproval;
 id | approver_id |          approval_type           |       approved_date        | notes 
----+-------------+----------------------------------+----------------------------+-------
  1 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 03:47:48.181347 | 
  2 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 03:47:55.808087 | 
  3 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 03:49:04.27095  | 
  4 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 03:49:05.46235  | 
  5 |           1 | ServiceKeyApprovalType.SUPERUSER | 2024-05-07 04:05:10.296796 | 
  6 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 04:43:57.099776 | 
  7 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 04:43:58.199101 | 
  8 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 04:43:58.740628 | 
  9 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 04:44:57.782562 | 
 10 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 06:04:46.102356 | 
 11 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 06:04:46.907342 | 
 12 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 06:15:00.004144 | 
 13 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 06:15:00.288843 | 
 14 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 06:26:13.871902 | 
 15 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 06:26:44.553869 | 
 16 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 06:26:44.980913 | 
 17 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 06:27:15.044647 | 
 18 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 06:46:41.631166 | 
 19 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 06:47:12.69024  | 
 20 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 06:47:12.72553  | 
 21 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 06:47:53.351615 | 
 22 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 07:04:05.20755  | 
 23 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 07:04:35.534945 | 
 24 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 07:04:36.499352 | 
 25 |             | ServiceKeyApprovalType.AUTOMATIC | 2024-05-07 07:05:05.599616 | 
(25 rows)
```

```
INSERT INTO servicekey (name, service, metadata, kid, jwk, created_date, expiration_date) 
  VALUES ('quay-readonly', 
                     'quay', 
                     '{}', 
                     'KVaF5mYCwo0iY1UTkq7Pl91oLE_yT9Upyo5Vl1Pb1K0', 
                     '{"n":"l5hXx6xvOL44uBI9uFtl0sOQclcyj7pLDXFdPN7kKUuuXa4Dw8L7ah0fPynNFN_0_KaE6xQ50nOpEcQSIspj0kOl2Nzpis2nWUr0-aNeWB87VlMsbNf5ZvrE36AGNdWVAi0pZucQhCjs5x0vQx2eG9o7X66frN8LnBjN9e71oH6Fo_oUIIwe4rT8ysBhOfri2qKs3AMIdDeB9uJ4E1-1CDax_bmzW7ERD1t8Pb2uC-FwsHvUHBI0myZCLyqu1o0kk6sbFJMMfufs1oSe4xrGjFmV77CMJMGrxPAAdNshd4Fy5-8xqr04LqOMvP7rGiIMU3swhgeexz911ybAQnbLfQ","e":"AQAB","kty":"RSA","kid":"KVaF5mYCwo0iY1UTkq7Pl91oLE_yT9Upyo5Vl1Pb1K0"}', 
                     '2023-06-10 12:10:10', 
                     '2024-06-10 12:10:10');
```

File: modules/optional-enabling-read-only-mode-backup-restore.adoc [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+[source,terminal]
+----
+$ base64 -w0 quay-config.yaml
+----
+
+. Scale down the {productname} Operator pods to `0`. This ensures that the Operator does not reconcile the secret after editing it.
++
+[source,terminal]
+----
+$ oc scale --replicas=0 deployment quay-operator.{productminv} -n openshift-operators
```
Comment (LiZhang19817) [inline modules/optional-enabling-read-only-mode-backup-restore.adoc:320]: **Here after updated the config secret managed by Quay Operator**, need to scale out Quay Operator, when Quay Customers want to put Quay to Normal State, remove following config from the config secret managed by Quay Operator, and let Quay Operator to reconcile the change.
```
REGISTRY_STATE: readonly
INSTANCE_SERVICE_KEY_KID_LOCATION: 'conf/stack/quay-readonly.kid'
INSTANCE_SERVICE_KEY_LOCATION: 'conf/stack/quay-readonly.pem'
```

File: modules/optional-enabling-read-only-mode-backup-restore.adoc [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+----
+INSERT 0 1
+----
+
+. Set the `approval_id` field on the created service key row to the `id` field from the created service key approval. You can use the following `SELECT` statements to get the necessary IDs:
++
+[source,terminal]
+----
+UPDATE servicekey 
+SET approval_id = (SELECT id FROM servicekeyapproval WHERE approval_type = 'Super User API')
```
Comment (LiZhang19817) [inline modules/optional-enabling-read-only-mode-backup-restore.adoc:179]: As the previous step insert approval type named **"ServiceKeyApprovalType.SUPERUSER"**, here should use "WHERE approval_type = 'ServiceKeyApprovalType.SUPERUSER'"

File: modules/optional-enabling-read-only-mode-backup-restore-standalone.adoc [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+----
+quay=# INSERT INTO servicekey 
+  ('name', 'service', 'metadata', 'kid', 'jwk', 'created_date', 'expiration_date')
+  VALUES ("quay-readonly",
+           "quay",
+           "{}",
+           "{<contents_of_.kid_file>}",
+           "{<contents_of_.jwk_file>}",
+           "{<created_date_of_read-only>}"
+           "{<expiration_date_of_read-only>}");
```
Comment (dongboyan77) [inline modules/optional-enabling-read-only-mode-backup-restore-standalone.adoc:146]: These values should use single quotes

File: modules/optional-enabling-read-only-mode-backup-restore-standalone.adoc [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+INSERT 0 1
+----
+
+. Next, add the key approval with the following query:
++
+[source,terminal]
+----
+quay=# INSERT INTO servicekeyapproval ('approval_type', 'notes')
+  VALUES ("ServiceKeyApprovalType.SUPERUSER",
+           {include_notes_here_on_why_this_is_being_added});
```
Comment (dongboyan77) [inline modules/optional-enabling-read-only-mode-backup-restore-standalone.adoc:162]: Need set 'approved_date' as well

File: modules/optional-enabling-read-only-mode-backup-restore-standalone.adoc [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+[source,terminal]
+----
+$ podman stop <quay_container_name_on_virtual_machine_b>
+----
+
+. Enter the following command to copy the contents of the `quay-readonly.kid` file and the `quay-readonly.pem` file to the directory that holds your {productname} configuration bundle:
++
+[source,terminal]
+----
+$ cp quay_readonly.kid quay-readonly.pem $Quay/config
```
Comment (dongboyan77) [inline modules/optional-enabling-read-only-mode-backup-restore-standalone.adoc:213]: typo, `quay-readonly.kid`

**Pull request merged**